VMRecovery/Testing/Create-TestVM.ps1

```diff
@@ -0,0 +1,848 @@
+        'RHEL' {$vmNamePrefix = 'rhel'}
+        'SLES' {$vmNamePrefix = 'sles'}
+        'openSUSE-Leap' {$vmNamePrefix = 'opensuse'}
+        'Oracle-Linux' {$vmNamePrefix = 'oraclelinux'}
+        'Datacenter-Core-1709' {$vmNamePrefix = 'rs3'}
+        'Datacenter-Core-1803' {$vmNamePrefix = 'rs4'}
+        '2019-datacenter' {$vmNamePrefix = 'rs5'}
+        '2016-Datacenter' {$vmNamePrefix = 'rs1'}
+        '2012-R2-Datacenter' {$vmNamePrefix = 'r212'}
+        '2012-Datacenter' {$vmNamePrefix = 'ws12'}
+        '2008-R2' {$vmNamePrefix = 'r208'}
+        'RS4-Pro' {$vmNamePrefix = 'rs4c'}
+        'RS3-Pro' {$vmNamePrefix = 'rs3c'}
+        'RS2-Pro' {$vmNamePrefix = 'rs2c'}
+        'Win7' {$vmNamePrefix = 'win7'}
+        'Win81' {$vmNamePrefix = 'win81'}
+        'Windows-10' {$vmNamePrefix = 'win10'}
+    }
+
+    # Some Linux sku names are the same between distros, e.g. 7.4 is a sku for both CentOS and Oracle-Linux, so use offer instead
+    switch -regex ($offer)
+    {
+        'UbuntuServer' {$vmNamePrefix = 'ubuntu'}
+        'CoreOS' {$vmNamePrefix = 'coreos'}
+        'Debian' {$vmNamePrefix = 'debian'}
+        'CentOS' {$vmNamePrefix = 'centos'}
+        'RHEL' {$vmNamePrefix = 'rhel'}
+        'SLES' {$vmNamePrefix = 'sles'}
+        'openSUSE-Leap' {$vmNamePrefix = 'opensuse'}
+        'Oracle-Linux' {$vmNamePrefix = 'oraclelinux'}
+    }
+
+    if($vmNamePrefix -eq ''){$vmNamePrefix = 'vm'}
+
+    #if (!$vmName -or $useDateInVMName)
+    if ($useDateInVMName)
+    {
+        $vmName = "$($vmNamePrefix)$(get-date (get-date).ToUniversalTime() -f MMddhhmmss)"
+    }
+    else
+    {
+        $vmName = $vmNamePrefix
+    }
+
+    return $vmName
+}
+
+function show-progress()
+{
+    param(
+        [string]$text,
+        [string]$prefix = 'timespan'
+    )
+
+    if ($prefix -eq 'timespan' -and $startTime)
+    {
+        $timespan = new-timespan -Start $startTime -End (get-date)
+        #$timespanString = '[{0:hh}:{0:mm}:{0:ss}.{0:ff}]' -f $timespan
+        $timespanString = '[{0:mm}:{0:ss}]' -f $timespan
+        write-host $timespanString -nonewline -ForegroundColor White
+        write-host " $text"
+    }
+    elseif ($prefix -eq 'both' -and $startTime)
+    {
+        $timestamp = get-date -format "yyyy-MM-dd hh:mm:ss"
+        $timespan = new-timespan -Start $startTime -End (get-date)
+        #$timespanString = "$($timestamp) $('[{0:hh}:{0:mm}:{0:ss}.{0:ff}]' -f $timespan)"
+        $timespanString = "$($timestamp) $('[{0:mm}:{0:ss}]' -f $timespan)"
+        write-host $timespanString -nonewline -ForegroundColor White
+        write-host " $text"
+    }
+    else
+    {
+        $timestamp = get-date -format "yyyy-MM-dd hh:mm:ss"
+        write-host $timestamp -nonewline -ForegroundColor Cyan
+        write-host " $text"
+    }
+}
+
+$PSDefaultParameterValues['*:ErrorAction'] = 'Stop'
+$PSDefaultParameterValues['*:WarningAction'] = 'SilentlyContinue'
+
+if ($location -eq 'francesouth'){exit}
```

Comment:
seems odd perhaps a comment here for special casing this region?

Reply:
Removed the testing folder